test/core.jl

```diff
@@ -4822,6 +4822,17 @@ end
 @test f14893() == 14893
 @test M14893.f14893() == 14893
 
+# issue #18725
+@test_nowarn begin
```

Comment:
Weird, right: the warning is getting output (correctly), but for some reason it is not captured by `redirect_stderr`.

For example, if I do the following in the test file:
```
let fname = tempname()
    try
        open(fname, "w") do f
            redirect_stderr(f) do
                begin
                    f18725x(x) = 1
                    f18725x(x) = 2
                    @test f18725x(0) == 2
                end
            end
        end
        println("GOT WARNING: \"", readstring(fname), '"')
    finally
        rm(fname, force=true)
    end
end
```
it outputs
```
WARNING: Method definition f18725x(Any) in module TestMain_core at /Users/stevenj/Documents/Code/julia/test/core.jl:4830 overwritten at /Users/stevenj/Documents/Code/julia/test/core.jl:4831.
GOT WARNING: ""
```
(And yet, the other `@test_warn` passes, indicating that the warning is *sometimes* being captured by `redirect_stderr`)

Any idea why the warning would not be captured here?